pallets/staking/src/pallet/impls.rs

```diff
@@ -888,29 +889,35 @@ impl<T: Config> Pallet<T> {
             };
 
             if let Some(Nominations { targets, .. }) = <Nominators<T>>::get(&voter) {
-                let voter_weight = weight_of(&voter);
-                if !targets.is_empty() {
-                    all_voters.push((voter.clone(), voter_weight, targets));
-                    nominators_taken.saturating_inc();
-                } else {
-                    // Technically should never happen, but not much we can do about it.
+                if Self::is_nominator_compliant(&voter) {
+                    let voter_weight = weight_of(&voter);
+                    if !targets.is_empty() {
+                        all_voters.push((voter.clone(), voter_weight, targets));
+                        nominators_taken.saturating_inc();
```

Comment:
I think we need to increment these `*_taken` counters even for filtered nominators/validators, since they are used to calculate the weight used by this method.